pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/errorprone/AvoidBranchingStatementAsLastInLoopRule.java

```diff
@@ -73,7 +75,7 @@ public Object visit(ASTBreakStatement node, Object data) {
 
     protected Object check(PropertyDescriptor<List<String>> property, Node node, Object data) {
         Node parent = node.getParent();
-        if (parent instanceof ASTBlock) {
+        while (parent instanceof ASTBlock || parent instanceof ASTFinallyClause || parent instanceof ASTTryStatement) {
```

Comment:
I think, this while loop produces some false positives now. E.g. the following should not be detected: https://github.com/spring-projects/spring-framework/blob/v5.3.13/spring-messaging/src/test/java/org/springframework/messaging/rsocket/RSocketBufferLeakTests.java#L262

We should only consider `break` when it is inside the finally clause. If it is inside the try block, it might not be executed when a exception occurs.

I'll push some changes to your branch...

Reply:
Oh, in PMD itself we use this pattern, which is now wrongly detected, why the build fails:

https://github.com/pmd/pmd/blob/e431831a6981b528108f7110da2dde544aaf9916/pmd-plsql/src/main/java/net/sourceforge/pmd/lang/plsql/symboltable/TypeSet.java#L159-L165